src/automaton/core/interop/ethereum/eth_helper_functions.h

```diff
@@ -125,7 +131,437 @@ static status eth_getTransactionReceipt(const std::string& url, const std::strin
   return curl_post(url, ss.str());
 }
 
+// Argument encoding and decoding
+
+static const uint32_t BUFFER_SIZE = 2048;
+
+enum abi_type {
+  empty = 0,
+  numerical = 1,  // bool, int<>, uint<>, fixed, unfixed, fixed<M>x<N>, ufixed<M>x<N>
```

Comment:
If these are not intended to be used in a combination, perhaps they don't need to be powers of two, but rather just a regular enum.

Reply:
They are intended to be used in a combination